src/Exceptions/InvalidWarehouseException.java

```diff
@@ -0,0 +1,7 @@
+package Exceptions;
+
+public class InvalidWarehouseException extends Exception {
```

Comment:
Più che invalid warehouse, la chiamerei EmptyWarehouseException. Il nome deve essere esplicito riguardo il problema. Invalid warehouse sembra più che la warehosue selezionata non sia corretta. Non so se condividi

Reply:
si capisco cosa intendi ma quella appunto è la condizione che invalida l'utilizzo del metodo, nel senso è un processo logico; se la wharehouse è vuota non è valida per la logica del metodo. al metodo cosa interessa di più ?penso avere un dato valido. A seconda di cosa torna isEmpty sappiamo se il magazzino è vuoto o meno, quando lanciamo l'eccezione tecnicamente credo vorremmo comunicare nello specifico che in questo caso il fatto che il magazzino è vuoto va ad invalidare tutta la logica del metodo e nel messaggio vai a specificare il motivo per il quale avviene ciò.

Reply:
"InvalidWarehouse" va bene come nome se utilizziamo quest'eccezione in vari casi per segnalare un problema nella warehouse, se però viene lanciata solo quando il magazzino è vuoto allora anche secondo me meglio optare per "EmptyWarehouse"

Reply:
perfetto, siccome questo task dovrebbe creare un'eccezione specifica per quel caso allora è meglio la emptyOne. Il task prevedeva unicamente la creazione dell'exception così da poter modificare i metodi che la richiedono integrandola